checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -323,28 +407,31 @@ private void inferOwningField(Node node, MethodInvocationNode invocation) {
   }
 
   /**
-   * Analyzes an assignment statement and performs three computations:
+   * Analyzes an assignment statement and performs the following computations. Does nothing if the
+   * rhs is not a local variable or parameter, or if the rhs has no must-call obligation.
    *
    * <ul>
+   *   <li>If the current method under analysis is a constructor, the left-hand side of the
+   *       assignment is the only owning field of the enclosing class, and the rhs is an alias of a
+   *       formal parameter, it adds an {@code @MustCallAlias} annotation to the formal parameter
+   *       and the result type of the constructor.
```

Comment:
Does this need to be re-done if some other part of the inference discovers an owning field?

Reply:
It won't be re-done in this iteration, in the next iteration it will infer an @Owning annotation if the number of owning fields is greater than 1. I added a new test file(`ReplaceMustCallAliasAnnotation.java`) to check this scenario.